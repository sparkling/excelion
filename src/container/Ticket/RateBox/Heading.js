import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {Type} from './index';

const StyledHeading = styled.div `
  display: flex;
  flex-direction: ${props => props.buySell === Type.BUY ? "row-reverse" : "row"};
  padding: 0.5em 0.75em;
  & > *:${props => props.buySell === Type.BUY ? "last" : "first"}-child {
    flex: 1;
  }
`

const MostSignificant = styled.div `
  font-size: 0.8em;
`
MostSignificant.displayName = 'MostSignificant';

const Action = styled.div `
  font-size: 0.8em
`
Action.displayName = 'Action';

const Heading = ({ rateMostSignificantDigits, buySell, currencySymbol }) =>
  (<StyledHeading buySell={buySell} >
    <Action>{buySell} {currencySymbol}</Action>
    <MostSignificant>{rateMostSignificantDigits}</MostSignificant>
  </StyledHeading>);

Heading.propTypes = {
  rateMostSignificantDigits: PropTypes.string.isRequired,
  buySell: PropTypes.string.isRequired,
  currencySymbol: PropTypes.string.isRequired,
}

export default Heading;
