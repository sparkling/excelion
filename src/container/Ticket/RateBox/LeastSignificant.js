import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledLeastSignificant = styled.div `
  text-align: center;
`

const Digit = styled.span `
  font-size: 5em;
  &:last-child {
    font-size: 2em;
  }
`
Digit.displayName='Digit';

const LeastSignificant = ({ rateLeastSignificantDigits }) => {
  const digits = rateLeastSignificantDigits.split("").map((digit, idx) => {
      return <Digit key={idx}>{digit}</Digit>;
    });

  return (<StyledLeastSignificant>{digits}</StyledLeastSignificant>);
}

LeastSignificant.propTypes = {
  rateLeastSignificantDigits: PropTypes.string.isRequired
}

export default LeastSignificant;
