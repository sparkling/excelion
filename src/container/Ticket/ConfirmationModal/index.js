import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter,
} from 'reactstrap';
import {Type} from '../RateBox';
import NumberFormat from 'react-number-format';

const ConfirmationModal = ({ type, amount, currency, clickRate, toggleModal, showModal }) =>
  (<Modal isOpen={showModal} toggle={toggleModal}>
    <ModalHeader toggle={toggleModal}>Receipt</ModalHeader>
    <ModalBody>
      <span>You {type === Type.BUY ? "bought" : "sold"} </span>
      <NumberFormat
          value={(amount * clickRate).toFixed(5)}
          displayType={'text'}
          thousandSeparator={true}
        />
        <span> {currency.source}</span>
    </ModalBody>
    <ModalFooter>
      <Button color="primary" onClick={toggleModal}>OK</Button>{' '}
    </ModalFooter>
  </Modal>);


ConfirmationModal.propTypes = {
  type: PropTypes.string,
  amount: PropTypes.number,
  currency: PropTypes.shape({
    source: PropTypes.string.isRequired,
  }),
  toggleModal: PropTypes.func.isRequired,
  clickRate: PropTypes.number,
}

export default ConfirmationModal;
