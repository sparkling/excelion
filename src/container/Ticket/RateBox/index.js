import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Heading from './Heading';
import LeastSignificant from './LeastSignificant';

export const Type = {
  "BUY": "BUY",
  "SELL": "SELL"
}

const Box = styled.div `
  font-family: arial;
  flex: 1;
  color: white;
  background-color: ${props => props.bgColor || "grey"};
`
Box.displayName = 'Box';

class RateBox extends React.PureComponent {

  constructor(props) {
    super(props);
    this.state = {
      bgColor: "grey",
    };
  }

  componentDidUpdate(prevProps) {
    const prevValue = parseInt(prevProps.rateLeastSignificantDigits, 10);
    const nextValue = parseInt(this.props.rateLeastSignificantDigits, 10);

    if (prevValue > nextValue) {
      this.setDecrease();
    }
    else if (prevValue < nextValue) {
      this.setIncrease();
    }
  }

  setIncrease() {
    this.setState(() => {
      return { bgColor: "blue"}
    });
  }

  setDecrease() {
    this.setState(() => {
      return { bgColor: "red"}
    });
  }

  render () {
    return (
      <Box bgColor={this.state.bgColor} onClick={this.props.onClick}>
        <Heading
          buySell={this.props.buySell}
          currencySymbol={this.props.currencySymbol}
          rateMostSignificantDigits={this.props.rateMostSignificantDigits}
        />
        <LeastSignificant
          rateLeastSignificantDigits={this.props.rateLeastSignificantDigits}
        />
      </Box>
    );
  }
}

RateBox.propTypes = {
  onClick: PropTypes.func.isRequired,
  rateMostSignificantDigits: PropTypes.string.isRequired,
  rateLeastSignificantDigits: PropTypes.string.isRequired,
  currencySymbol: PropTypes.string.isRequired,
  buySell: PropTypes.string.isRequired,
}

export default RateBox;
