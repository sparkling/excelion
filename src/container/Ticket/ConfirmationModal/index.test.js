import React from 'react';
import 'jest-styled-components';
import ConfirmationModal from './index';
import { Type } from '../RateBox';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('<ConfirmationModal />', () => {

  it('should render the same buy snapshot', () => {
    const wrapper = mount(
      <ConfirmationModal
        type={Type.BUY}
        amount={10}
        currency={{"source":"EUR"}}
        toggleModal={()=>{}}
        showModal={true}
        clickRate={1.40123}
      /> );

      expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('should render the same sell snapshot', () => {
    const wrapper = mount(
      <ConfirmationModal
        type={Type.SELL}
        amount={10}
        currency={{"source":"EUR"}}
        toggleModal={()=>{}}
        showModal={true}
        clickRate={1.40123}
      /> );

      expect(toJson(wrapper)).toMatchSnapshot()
  });

});
