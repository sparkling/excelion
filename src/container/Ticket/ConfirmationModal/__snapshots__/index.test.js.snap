// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationModal /> should render the same buy snapshot 1`] = `
<ConfirmationModal
  amount={10}
  clickRate={1.40123}
  currency={
    Object {
      "source": "EUR",
    }
  }
  showModal={true}
  toggleModal={[Function]}
  type="BUY"
>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <Portal
      node={
        <div
          tabindex="-1"
          style="position: relative; z-index: 1050;"
        >
          <div
            class=""
          >
            <div
              class="modal fade"
              style="display: block;"
              role="dialog"
              tabindex="-1"
            >
              <div
                class="modal-dialog"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal-header"
                  >
                    <h5
                      class="modal-title"
                    >
                      Receipt
                    </h5>
                    <button
                      type="button"
                      class="close"
                      aria-label="Close"
                    >
                      <span
                        aria-hidden="true"
                      >
                        ×
                      </span>
                    </button>
                  </div>
                  <div
                    class="modal-body"
                  >
                    <span>
                      You 
                      bought
                    </span>
                    <span>
                      14.01230
                    </span>
                    <span>
                      EUR
                    </span>
                  </div>
                  <div
                    class="modal-footer"
                  >
                    <button
                      type="button"
                      class="btn btn-primary"
                    >
                      OK
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="modal-backdrop fade"
            />
          </div>
        </div>
      }
    >
      <Portal
        containerInfo={
          <div
            tabindex="-1"
            style="position: relative; z-index: 1050;"
          >
            <div
              class=""
            >
              <div
                class="modal fade"
                style="display: block;"
                role="dialog"
                tabindex="-1"
              >
                <div
                  class="modal-dialog"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <h5
                        class="modal-title"
                      >
                        Receipt
                      </h5>
                      <button
                        type="button"
                        class="close"
                        aria-label="Close"
                      >
                        <span
                          aria-hidden="true"
                        >
                          ×
                        </span>
                      </button>
                    </div>
                    <div
                      class="modal-body"
                    >
                      <span>
                        You 
                        bought
                      </span>
                      <span>
                        14.01230
                      </span>
                      <span>
                        EUR
                      </span>
                    </div>
                    <div
                      class="modal-footer"
                    >
                      <button
                        type="button"
                        class="btn btn-primary"
                      >
                        OK
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="modal-backdrop fade"
              />
            </div>
          </div>
        }
      >
        <div
          className=""
        >
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onClick={[Function]}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            role="dialog"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex="-1"
            tag="div"
            timeout={300}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={300}
              unmountOnExit={false}
            >
              <div
                className="modal fade"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                role="dialog"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <ModalHeader
                      charCode={215}
                      closeAriaLabel="Close"
                      tag="h5"
                      toggle={[Function]}
                      wrapTag="div"
                    >
                      <div
                        className="modal-header"
                      >
                        <h5
                          className="modal-title"
                        >
                          Receipt
                        </h5>
                        <button
                          aria-label="Close"
                          className="close"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                        </button>
                      </div>
                    </ModalHeader>
                    <ModalBody
                      tag="div"
                    >
                      <div
                        className="modal-body"
                      >
                        <span>
                          You 
                          bought
                           
                        </span>
                        <NumberFormat
                          allowEmptyFormatting={false}
                          allowNegative={true}
                          decimalSeparator="."
                          displayType="text"
                          fixedDecimalScale={false}
                          getInputRef={[Function]}
                          isAllowed={[Function]}
                          isNumericString={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onMouseUp={[Function]}
                          onValueChange={[Function]}
                          prefix=""
                          suffix=""
                          thousandSeparator={true}
                          type="text"
                          value="14.01230"
                        >
                          <span>
                            14.01230
                          </span>
                        </NumberFormat>
                        <span>
                           
                          EUR
                        </span>
                      </div>
                    </ModalBody>
                    <ModalFooter
                      tag="div"
                    >
                      <div
                        className="modal-footer"
                      >
                        <Button
                          color="primary"
                          onClick={[Function]}
                          tag="button"
                        >
                          <button
                            className="btn btn-primary"
                            onClick={[Function]}
                            type="button"
                          >
                            OK
                          </button>
                        </Button>
                         
                      </div>
                    </ModalFooter>
                  </div>
                </div>
              </div>
            </Transition>
          </Fade>
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal-backdrop"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={150}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={150}
              unmountOnExit={false}
            >
              <div
                className="modal-backdrop fade"
              />
            </Transition>
          </Fade>
        </div>
      </Portal>
    </Portal>
  </Modal>
</ConfirmationModal>
`;

exports[`<ConfirmationModal /> should render the same sell snapshot 1`] = `
<ConfirmationModal
  amount={10}
  clickRate={1.40123}
  currency={
    Object {
      "source": "EUR",
    }
  }
  showModal={true}
  toggleModal={[Function]}
  type="SELL"
>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <Portal
      node={
        <div
          tabindex="-1"
          style="position: relative; z-index: 1050;"
        >
          <div
            class=""
          >
            <div
              class="modal fade"
              style="display: block;"
              role="dialog"
              tabindex="-1"
            >
              <div
                class="modal-dialog"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal-header"
                  >
                    <h5
                      class="modal-title"
                    >
                      Receipt
                    </h5>
                    <button
                      type="button"
                      class="close"
                      aria-label="Close"
                    >
                      <span
                        aria-hidden="true"
                      >
                        ×
                      </span>
                    </button>
                  </div>
                  <div
                    class="modal-body"
                  >
                    <span>
                      You 
                      sold
                    </span>
                    <span>
                      14.01230
                    </span>
                    <span>
                      EUR
                    </span>
                  </div>
                  <div
                    class="modal-footer"
                  >
                    <button
                      type="button"
                      class="btn btn-primary"
                    >
                      OK
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="modal-backdrop fade"
            />
          </div>
        </div>
      }
    >
      <Portal
        containerInfo={
          <div
            tabindex="-1"
            style="position: relative; z-index: 1050;"
          >
            <div
              class=""
            >
              <div
                class="modal fade"
                style="display: block;"
                role="dialog"
                tabindex="-1"
              >
                <div
                  class="modal-dialog"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <h5
                        class="modal-title"
                      >
                        Receipt
                      </h5>
                      <button
                        type="button"
                        class="close"
                        aria-label="Close"
                      >
                        <span
                          aria-hidden="true"
                        >
                          ×
                        </span>
                      </button>
                    </div>
                    <div
                      class="modal-body"
                    >
                      <span>
                        You 
                        sold
                      </span>
                      <span>
                        14.01230
                      </span>
                      <span>
                        EUR
                      </span>
                    </div>
                    <div
                      class="modal-footer"
                    >
                      <button
                        type="button"
                        class="btn btn-primary"
                      >
                        OK
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="modal-backdrop fade"
              />
            </div>
          </div>
        }
      >
        <div
          className=""
        >
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onClick={[Function]}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            role="dialog"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex="-1"
            tag="div"
            timeout={300}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={300}
              unmountOnExit={false}
            >
              <div
                className="modal fade"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                role="dialog"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <ModalHeader
                      charCode={215}
                      closeAriaLabel="Close"
                      tag="h5"
                      toggle={[Function]}
                      wrapTag="div"
                    >
                      <div
                        className="modal-header"
                      >
                        <h5
                          className="modal-title"
                        >
                          Receipt
                        </h5>
                        <button
                          aria-label="Close"
                          className="close"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                        </button>
                      </div>
                    </ModalHeader>
                    <ModalBody
                      tag="div"
                    >
                      <div
                        className="modal-body"
                      >
                        <span>
                          You 
                          sold
                           
                        </span>
                        <NumberFormat
                          allowEmptyFormatting={false}
                          allowNegative={true}
                          decimalSeparator="."
                          displayType="text"
                          fixedDecimalScale={false}
                          getInputRef={[Function]}
                          isAllowed={[Function]}
                          isNumericString={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onMouseUp={[Function]}
                          onValueChange={[Function]}
                          prefix=""
                          suffix=""
                          thousandSeparator={true}
                          type="text"
                          value="14.01230"
                        >
                          <span>
                            14.01230
                          </span>
                        </NumberFormat>
                        <span>
                           
                          EUR
                        </span>
                      </div>
                    </ModalBody>
                    <ModalFooter
                      tag="div"
                    >
                      <div
                        className="modal-footer"
                      >
                        <Button
                          color="primary"
                          onClick={[Function]}
                          tag="button"
                        >
                          <button
                            className="btn btn-primary"
                            onClick={[Function]}
                            type="button"
                          >
                            OK
                          </button>
                        </Button>
                         
                      </div>
                    </ModalFooter>
                  </div>
                </div>
              </div>
            </Transition>
          </Fade>
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal-backdrop"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={150}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={150}
              unmountOnExit={false}
            >
              <div
                className="modal-backdrop fade"
              />
            </Transition>
          </Fade>
        </div>
      </Portal>
    </Portal>
  </Modal>
</ConfirmationModal>
`;
